deepchecks/vision/utils/transformations.py

```diff
@@ -9,21 +9,50 @@
 # ----------------------------------------------------------------------------
 #
 """Module for defining functions related to vision transforms."""
+import abc
+import typing as t
 from copy import copy
-from typing import Optional, Sized
 
 import albumentations
 import imgaug.augmenters as iaa
+import numpy as np
 import torch
+import torchvision.transforms as T
+from PIL import Image
 
 from deepchecks.core.errors import (DeepchecksNotSupportedError,
                                     DeepchecksValueError)
+from deepchecks.vision.vision_data import TaskType
 
-__all__ = ['get_transforms_handler', 'add_augmentation_in_start', 'un_normalize_batch',
+__all__ = ['get_transforms_handler', 'un_normalize_batch', 'AbstractTransformations',
            'ImgaugTransformations', 'AlbumentationsTransformations']
 
 
-class ImgaugTransformations:
+class AbstractTransformations(abc.ABC):
+    """Abstract class for supporting functions for various transforms."""
+
+    is_image_shifting = True
```

Comment:
@noamzbr is "shifting" the only transformation that affects labels? I wonder if this should be called "is_transforming_labels" or something

Reply:
yeah agree is_transforming_labels is better